compiler.go

```diff
@@ -430,6 +474,17 @@ func (t *task) asFile(ctx context.Context, name string, r SearchResult) (linker.
 				return nil, ctx.Err()
 			}
 		}
+		if descriptorProtoRes != nil {
+			select {
+			case <-descriptorProtoRes.ready:
+				// descriptor.proto wasn't explicitly imported, so we can ignore a failure
+				if descriptorProtoRes.err == nil {
```

Comment:
I walked through this in the debugger and saw one non-nil error reported in the linker tests - `search result for "google/protobuf/descriptor.proto" returned descriptor for "foo.proto"`. Is that expected?

Reply:
Possibly. What test? If you look at the resolver it's using, it may be hard-coded to return a particular descriptor, regardless of the requested file name.

In any event, since this is not an explicit dependency, I don't think we should fail here if the resolver can't supply something. If the _explicitly_ import this file, then it will fail as expected.

Reply:
I'll see if I can gather more details. I agree with the behavior just wanted to make sure there wasn't something we were missing.